bad_message_wizard.py

```diff
@@ -150,28 +150,49 @@ def reset_bad_message_cache():
 
 
 def show_bad_message_list():
-    bad_messages = list_bad_messages()
+    bad_messages: list = list_bad_message_summaries()
     if not bad_messages:
         show_no_bad_messages()
         return
+    bad_messages.sort(key=lambda message: message['firstSeen'])
     print('')
     print(f'There are currently {len(bad_messages)} bad messages:')
     if len(bad_messages) > 20:
-        print('Showing only the first 20')
+        print('Showing only the oldest 20')
         bad_messages = bad_messages[:20]
 
-    for index, bad_message in enumerate(bad_messages):
-        print(f'  {colored(f"{index + 1}.", "white")} {bad_message}')
+    pretty_print_bad_message_summaries(bad_messages)
     print('')
 
-    raw_selection = input(colored(f'Select a message (1 to {len(bad_messages)}) or cancel with ENTER): ', 'white'))
+    raw_selection = input(colored(f'Select a message (1 to {len(bad_messages)}) or cancel with ENTER: ', 'cyan'))
     print('')
 
     valid_selection = validate_integer_input_range(raw_selection, 1, len(bad_messages))
     if not valid_selection:
         return
 
-    show_bad_message_metadata_for_hash(bad_messages[valid_selection - 1])
+    in_bad_message_context = True
+    while in_bad_message_context:
+        in_bad_message_context = show_bad_message_metadata_for_hash(bad_messages[valid_selection - 1]['messageHash'])
+
+
+def pretty_print_bad_message_summaries(bad_message_summaries):
+    column_widths = {
+        'messageHash': len(bad_message_summaries[0]['messageHash']),
+        'firstSeen': max(len(str(summary['firstSeen'])) for summary in bad_message_summaries),
+        'quarantined': len("Quarantined"),
+        'queues': max(len(str(summary['affectedQueues'])) for summary in bad_message_summaries),
+    }
+    print(f'      | {colored("Message Hash".ljust(column_widths["messageHash"]), color="cyan")} '
+          f'| {colored("First Seen".ljust(column_widths["firstSeen"]), color="cyan")} '
+          f'| {colored("Quarantined", color="cyan")} '
+          f'| {colored("Queues".ljust(column_widths["queues"]), color="cyan")}')
+    for index, summary in enumerate(bad_message_summaries, 1):
+        print(f'   {colored(str(index).ljust(2), color="cyan")} '
+              f'| {summary["messageHash"]} '
+              f'| {summary["firstSeen"]} '
+              f'| {str(summary["quarantined"]).ljust(column_widths["quarantined"])} '
+              f'| {summary["affectedQueues"]}')
```

Comment:
Could put in the `affectedServices` here as well? Or would that mean too many columns..?